apisix/plugins/prometheus/exporter.lua

```diff
@@ -351,6 +356,42 @@ local function etcd_modify_index()
 
 end
 
+local key = {}
+local combine_param = {}
+
+local function shared_dict_status()
+    local header_of_shared_dict = core.request.header(ngx.ctx, "shared_dict")
```

Comment:
People don't use Prometheus in that way. Usually the application developers decide what metrics will be exposed via their local configuration, but not Prometheus's configuration.

Reply:
Not even said, the Prometheus doesn't support custom header AFAIK.